### PR TITLE
enh(Navigation): Close-menu-when-a-navigation-item-is-clicked

### PR DESCRIPTION
**Pull request by Noha-ElAbrouki**
close menu when navigation item is clicked

### CODE REVIEW

File: www/front_src/src/Navigation/Sidebar/Menu/index.tsx [hunk 1]
```diff
@@ -214,20 +218,15 @@ const NavigationMenu = ({
...
 
-      addSelectedNavigationItemsByDefault(filteredResult);
-    });
-  }, []);
+  const leaveMenuItem = (): void => {
+    setIsDoubleClickedFromRoot(false);
+  };
 
-  useEffect(() => {
+  const defaultItems = (): void => {
```
Comment (Thebarda) [inline www/front_src/src/Navigation/Sidebar/Menu/index.tsx:229]:
```suggestion
  const setDefaultItems = (): void => {
```

File: www/front_src/src/Navigation/Sidebar/Menu/index.tsx [hunk 2]
```diff
@@ -238,6 +237,16 @@ const NavigationMenu = ({
...
+  };
+
+  document.addEventListener('visibilitychange', () => {
+    if (equals(document.visibilityState, 'visible')) {
+      defaultItems();
+    }
+  });
+
+  useEffect(() => {
+    defaultItems();
```
Comment (Thebarda) [inline www/front_src/src/Navigation/Sidebar/Menu/index.tsx:249]:
```suggestion
      setDefaultItems();
    }
  });

  useEffect(() => {
    setDefaultItems();
```

File: www/front_src/src/Navigation/Sidebar/Menu/index.tsx [hunk 3]
```diff
@@ -214,20 +218,15 @@ const NavigationMenu = ({
...
 
-  useEffect(() => {
-    navigationData?.forEach((item) => {
-      const searchedItems = searchItemsHoveredByDefault(item);
-      const filteredResult = flatten(searchedItems || []).filter(Boolean);
-
-      if (isEmpty(filteredResult)) {
-        return;
-      }
+  const itemNavigated = (): void => {
```
Comment (Thebarda) [inline www/front_src/src/Navigation/Sidebar/Menu/index.tsx:221]:
```suggestion
  const collapseMenu= (): void => {
```

File: www/front_src/src/Navigation/Sidebar/Menu/index.tsx [hunk 4]
```diff
@@ -292,6 +303,7 @@ const NavigationMenu = ({
                     {...props}
                     data={item.children}
                     isCollapsed={index === hoveredIndex}
+                    isNavigated={itemNavigated}
```
Comment (Thebarda) [inline www/front_src/src/Navigation/Sidebar/Menu/index.tsx:306]:
```suggestion
                    collapseMenu={collapseMenu}
```

File: www/front_src/src/Navigation/Sidebar/Menu/index.tsx [hunk 5]
```diff
@@ -72,7 +74,7 @@ const NavigationMenu = ({
 
   const levelName = 'level_0';
   const currentWidth = isDrawerOpen ? openedDrawerWidth / 8 : closedDrawerWidth;
-
+  const dismissDurationMenu = theme.transitions.duration.complex;
```
Comment (Thebarda) [inline www/front_src/src/Navigation/Sidebar/Menu/index.tsx:77]:
```suggestion
  const dismissMenuDuration = theme.transitions.duration.complex;
  
```

File: www/front_src/src/Navigation/Sidebar/Menu/index.tsx [hunk 6]
```diff
@@ -94,13 +96,15 @@ const NavigationMenu = ({
     timeoutRef.current = setTimeout((): void => {
       setHoveredIndex(null);
       setHoveredNavigationItems(null);
-    }, 500);
+    }, dismissDurationMenu);
```
Comment (Thebarda) [inline www/front_src/src/Navigation/Sidebar/Menu/index.tsx:99]:
```suggestion
    }, dismissMenuDuration);
```

File: www/front_src/src/Navigation/Sidebar/Menu/index.tsx [hunk 7]
```diff
@@ -214,20 +218,15 @@ const NavigationMenu = ({
...
+  const collapseMenu = (): void => {
+    setHoveredIndex(null);
+  };
 
-      addSelectedNavigationItemsByDefault(filteredResult);
-    });
-  }, []);
+  const leaveMenuItem = (): void => {
+    setIsDoubleClickedFromRoot(false);
+  };
```
Comment (Thebarda) [inline www/front_src/src/Navigation/Sidebar/Menu/index.tsx:227]:
```suggestion
  };
  
```

**Review by kduret — COMMENTED** [review]
does it fix this issue too ? https://www.loom.com/share/85530e22d81943bdb38c1889467a7f41
(when level 3 is hovered very quickly)

**Review by Thebarda — APPROVED** [review]

**Review by bdauria — APPROVED** [review]

**Pull request merged**